tests/functional/test_thermal_heuristic_two_clusters_with_bc.py

```diff
@@ -0,0 +1,273 @@
+        output_idx=ExpectedOutputIndexes(
+            idx_generation=4, idx_nodu=6, idx_spillage=29, idx_unsupplied=25
+        ),
+    )
+    expected_output.check_output_values(main_resolution_step.output)
+
+
+def test_accurate_heuristic(
+    data_path: str,
+    models: list[Model],
+    week_scenario_index: WeekScenarioIndex,
+) -> None:
+    """
+    Solve the same problem as before with the heuristic accurate of Antares. The accurate heuristic decides to turn on 3 units of the first cluster but due to the binding constraint and p_min, the problem become infeasible.
+    """
+
+    number_hours = 168
+    thermal_problem_builder = ThermalProblemBuilder(
+        fast=False,
+        data_dir=Path(__file__).parent / data_path,
+        id_thermal_cluster_model=THERMAL_CLUSTER_MODEL_MILP.id,
+        port_types=[BALANCE_PORT_TYPE],
+        models=[AccurateModelBuilder(THERMAL_CLUSTER_MODEL_MILP).model] + models,
+        time_scenario_hour_parameter=TimeScenarioHourParameter(1, 1, number_hours),
+    )
+
+    # First optimization
+    resolution_step_1 = thermal_problem_builder.main_resolution_step(
+        week_scenario_index
+    )
+
+    # Get number of on units and round it to integer
+    thermal_problem_builder.update_database_heuristic(
+        resolution_step_1.output,
+        week_scenario_index,
+        None,
+        param_to_update="nb_units_min",
+        var_to_read="nb_on",
+        fn_to_apply=lambda x: ceil(round(x, 12)),
+    )
+
+    for g in thermal_problem_builder.heuristic_components():
+        for time_step in range(number_hours):
+            assert thermal_problem_builder.database.get_value(
+                ComponentParameterIndex(g, "nb_units_min"), time_step, 0
+            ) == (2 if time_step != 12 and g == "G1" else (3 if g == "G1" else 0))
+
+        # Solve heuristic problem
+        resolution_step_accurate_heuristic = (
+            thermal_problem_builder.heuristic_resolution_step(
+                week_scenario_index,
+                id_component=g,
+                model=HeuristicAccurateModelBuilder(THERMAL_CLUSTER_MODEL_MILP).model,
+            )
+        )
+
+        thermal_problem_builder.update_database_heuristic(
+            resolution_step_accurate_heuristic.output,
+            week_scenario_index,
+            [g],
+            param_to_update="nb_units_min",
+            var_to_read="nb_on",
+            fn_to_apply=lambda x: ceil(round(x, 12)),
+        )
+
+        for time_step in range(number_hours):
+            assert thermal_problem_builder.database.get_value(
+                ComponentParameterIndex(g, "nb_units_min"), time_step, 0
+            ) == (2 if time_step != 12 and g == "G1" else (3 if g == "G1" else 0))
+
+    # Second optimization with lower bound modified
+    resolution_step_2 = thermal_problem_builder.main_resolution_step(
+        week_scenario_index,
+        SolvingParameters(expected_status=pywraplp.Solver.INFEASIBLE),
```

Comment:
Here we see that you need to separate the building of the problem (done in the class ThermalProblemBuilder), the resolution (from OptimizationProblem) and the check of the status (class ExpectedOutput/Status or just a simple assert without need of a class